v8_migration_guide.md

```diff
@@ -1,30 +1,39 @@
- incomplete deserialized object or runtime exceptions. It is necessary that you upgrade to the 
- latest API version to ensure events are created with the same API version supported by this library. 
- 
+### Version 8 now has a pinned API version, and model classes are no longer backward-compatible
+Up until `stripe-java` version 7, model classes had been maintained in a backward-compatible way.
+Regardless of your [Stripe API version](https://dashboard.stripe.com/developers), you could 
+always use the latest version of this client library. Starting with `stripe-java` 8.0.0, model 
+classes strictly describe schema of API responses from a specific API version. Each library 
+release now specifies an API version it supports: API version is pinned in the library. This 
+first version 8.0.0 has the pinned version of [2019-03-14](https://stripe.com/docs/upgrades#2019-03-14). 
+
+### Remaining on an old API version can cause deserialization failure on webhook events.
+Under the hood, new version 8 and above sets `Stripe-Version` request header to 
+`Stripe#API_VERSION` statically-defined in the library. As a result, making API request is
+safe, because the return API response has schema of the pinned version and is compatible with the 
+library's Java model classes. However, webhook events are by default rendered at your 
+account API version and maybe incompatible with the library's schema, causing deserialization 
+failures.
+
+However, you can ensure safe serialization of incoming events by [creating live webhook events](https://stripe.com/docs/api/webhook_endpoints/create#create_webhook_endpoint-api_version) 
+to this pinned API version. Though, with such integration ready, you will be effectively prepared 
+to do an API version upgrade. In fact, configuring test/live webhook events to a new API version
+is a recommended path to do a safe version upgrade. 
+
+### Support for reading old events `/v1/events/*` via `Event#retrieve`
+Events can be retrieved via standard API request, but unlike normal API response, the event data 
+is not rendered at the version specified by request header. Specifically, the event data object
+in `Event#data#object` is by default rendered according to your Stripe API version the time of 
+its creation. (The outer event structure describing event's metadata though does follow the 
+request header version.) Version 8 supports a custom-defined compatibility layer that you can 
+patch events or implement a best-attempt getter, instead of failing deserialization upon seeing 
+events with incompatible schema. This will be discussed in detail in the later section.
+
 ## API version upgrade with `stripe-java`
+Using `stripe-java` version 8 and above means that your API requests will behave as if you have 
+upgraded to the pinned API version. To handle this change, the work required is 
+actually the same as when you would do an actual API version upgrade on your account. 
```

Comment:
I qualify the reason to follow version upgrade here on the actual work being the same.